tests/php/src/BlockUniqidClassNameTransformerTest.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace AmpProject\AmpWP\Tests;
+
+use AMP_Options_Manager;
+use AMP_Theme_Support;
+use AmpProject\AmpWP\BlockUniqidClassNameTransformer;
+use AmpProject\AmpWP\Infrastructure\Conditional;
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use AmpProject\AmpWP\Option;
+use AmpProject\AmpWP\Tests\Helpers\MarkupComparison;
+
+/** @coversDefaultClass \AmpProject\AmpWP\BlockUniqidClassNameTransformer */
+final class BlockUniqidClassNameTransformerTest extends TestCase {
+
+	use MarkupComparison;
+
+	/** @var BlockUniqidClassNameTransformer */
+	private $instance;
+
+	/**
+	 * Setup.
+	 *
+	 * @inheritdoc
+	 */
+	public function setUp() {
+		parent::setUp();
+		$this->instance = new BlockUniqidClassNameTransformer();
+	}
+
+	public function test_it_can_be_initialized() {
+		$this->assertInstanceOf( Conditional::class, $this->instance );
+		$this->assertInstanceOf( Delayed::class, $this->instance );
+		$this->assertInstanceOf( Registerable::class, $this->instance );
+		$this->assertInstanceOf( Service::class, $this->instance );
+	}
+
+	/** @covers ::get_registration_action() */
+	public function test_get_registration_action() {
+		$this->assertEquals( 'wp', BlockUniqidClassNameTransformer::get_registration_action() );
+	}
+
+	/** @covers ::is_needed() */
+	/** @covers ::check_core_version() */
+	/** @covers ::has_gutenberg_plugin() */
+	public function test_is_needed() {
+		if (
+			version_compare( get_bloginfo( 'version' ), '5.9', '<' )
+			&&
+			! defined( 'GUTENBERG_VERSION' )
+		) {
+			$this->assertFalse( BlockUniqidClassNameTransformer::is_needed() );
+		} else {
+			AMP_Options_Manager::update_option( Option::THEME_SUPPORT, AMP_Theme_Support::READER_MODE_SLUG );
+			$post_id = self::factory()->post->create();
+
+			$this->go_to( get_permalink( $post_id ) );
+			$this->assertFalse( amp_is_request() );
+			$this->assertFalse( BlockUniqidClassNameTransformer::is_needed() );
+
+			$this->go_to( amp_get_permalink( $post_id ) );
+			$this->assertTrue( amp_is_request() );
+			$this->assertTrue( BlockUniqidClassNameTransformer::is_needed() );
+		}
+	}
+
+	/** @covers ::register() */
+	/** @covers ::get_block_inline_style_registration_hook_name() */
+	public function test_register() {
+		$this->instance->register();
+		$this->assertEquals( PHP_INT_MAX, has_filter( 'render_block', [ $this->instance, 'transform_class_names_in_block_content' ] ) );
+
+		$expected_hook_name = function_exists( 'wp_is_block_theme' ) && wp_is_block_theme() ? 'wp_enqueue_scripts' : 'wp_footer';
```

Comment:
This isn't good but it seems that mocking the `wp_is_block_theme` function would be pretty hard (there are no hooks that would allow us to change the output of this function).